app/models/message_notifier.rb

```diff
@@ -0,0 +1,90 @@
+class MessageNotifier
+  def initialize(user, ability, facility)
+    @user = user
+    @ability = ability
+    @facility = facility
+  end
+
+  def message_count
+    notifications.count +
+    order_details_in_dispute.count +
+    problem_order_details.count +
+    problem_reservation_order_details.count +
+    training_requests.count
+  end
+
+  def messages?
+    notifications? ||
```

Comment:
I'm not sure I'm convinced that the extra predicate method wrapping `any?` on each of these is adding much value. If we're not wrapping the `count`s above, maybe we don't wrap them here either.